File: shoppingApp/backend/shoppingApp/src/main/java/com/example/ShoppingApp/model/Product.java
package com.example.ShoppingApp.model;

import java.sql.Blob;
import java.util.ArrayList;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import com.example.ShoppingApp.entity.AdminEntity;
import com.example.ShoppingApp.entity.ShoppingCartEntity;

public class Product {
	
	private int id;
	private String name;
	private String category;
	private int quantity;
	private String condition;
	private double price;
	private byte[] imgURL;

	private List<ShoppingCartQuantityEntity> ShoppingCartQuantityEntity = new ArrayList<>();

	private AdminEntity adminEntity;


	public double getPrice() {
		return price;
	}
	public void setPrice(double price) {
		this.price = price;
	}
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public String getCategory() {
		return category;
	}
	public void setCategory(String category) {
		this.category = category;
	}
	public int getQuantity() {
		return quantity;
	}
	public void setQuantity(int quantity) {
		this.quantity = quantity;
	}
	public String getCondition() {
		return condition;
	}
	public void setCondition(String condition) {
		this.condition = condition;
	}

	public byte[] getImgURL() {
		return imgURL;
	}
	public void setImgURL(byte[] imgURL) {
		this.imgURL = imgURL;
	}
	public List<ShoppingCartQuantityEntity> getShoppingCartQuantityEntity() {
		return ShoppingCartQuantityEntity;
	}
	public void setShoppingCartQuantityEntity(List<ShoppingCartQuantityEntity> shoppingCartQuantityEntity) {
		ShoppingCartQuantityEntity = shoppingCartQuantityEntity;
	}
	public AdminEntity getAdminEntity() {
		return adminEntity;
	}
	public void setAdminEntity(AdminEntity adminEntity) {
		this.adminEntity = adminEntity;
	}

	

}

File: shoppingApp/frontend/shoppingApp/src/app/product.ts
export class Product {
    id:number;
    name:string;
    category:string;
    quantity:number;
    condition:string;
    price:number;
    imgURL:any;

    constructor(
        id:number,
        name:string,
        category:string,
        quantity:number,
        condition:string,
        price:number,
        imgURL:any,

    ){
        this.id=id;
        this.name=name;
        this.category=category;
        this.quantity=quantity;
        this.condition=condition;
        this.price=price;
        this.imgURL=imgURL;
    }
    public get _id(): number {
        return this.id;
    }
    public set _id(id: number) {
        this.id = id;
    }

    public get _name(): string {
        return this.name;
    }
    public set _name(name: string) {
        this.name = name;
    }

    public get _category(): string {
        return this.category;
    }
    public set _category(category: string) {
        this.category = category;
    }

    public get _quantity(): number {
        return this.quantity;
    }
    public set _quantity(quantity: number) {
        this.quantity = quantity;
    }

    public get _price(): number {
        return this.price;
    }
    public set _price(price: number) {
        this.price = price;
    }

    public get _condition(): string {
        return this.category;
    }
    public set _condition(condition: string) {
        this.condition = condition;
    }

    public get _imgURL(): any {
        return this.category;
    }
    public set _imgURL(imgURL: any) {
        this.imgURL = imgURL;
    }

}
